llvm/lib/ProfileData/InstrProf.cpp

```diff
@@ -560,6 +602,28 @@ Error InstrProfSymtab::addFuncWithName(Function &F, StringRef PGOFuncName) {
   return Error::success();
 }
 
+uint64_t InstrProfSymtab::getVTableHashFromAddress(uint64_t Address) {
+  finalizeSymtab();
```

Comment:
Like the lookup for GlobalVariable, why do we need to call finalizeSymtab here?

Reply:
Added a comment.

Thinking about it, there might be a way to call 'finalize' once to make the code simpler (illustrated below). If the overall idea looks good, I could work on a separate change to refactor existing code.

'Symtab' implementation is something like this

```
Status Symtab::addElement() {
   if (finalized)
     return error("symtab cannot have new element after being finalized");
  // add element implementation
}

void Symtab::finalize() {
  llvm::sort(vectors);
  finalized=true;
}
```

And user code is like

```
auto symtab = createSymtab();

for (auto element : elements) {
  symtab.addElement(element);
}

symtab.finalize();

auto result = symtab.getFuncHashFromAddress(addr);
...
```

Reply:
Summarizing offline discussion - if we chose to refactor it makes more sense to add appropriate factory methods of cases `addElement` are called. This seems to be the intent of the original implementation with a few different `create` methods. Then we can make `addElement` private. This will enforce what we want through a better API rather than runtime checks. 

For this patch, we can add a comment and look into the refactor later.